src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -281,12 +282,14 @@ class NexusPublishPlugin : Plugin<Project> {
     private fun getRepoUrl(
         nexusRepo: NexusRepository,
         extension: NexusPublishExtension,
-        registry: Provider<StagingRepositoryDescriptorRegistry>,
-        provideFallback: Boolean
+        registry: Provider<InvalidatingStagingRepositoryDescriptorRegistry>,
+        provideFallback: Boolean,
+        artifactRepo: ArtifactRepository
     ) = if (extension.useStaging.get()) {
         val descriptorRegistry = registry.get()
         if (provideFallback) {
-            descriptorRegistry.tryGet(nexusRepo.name)?.stagingRepositoryUrl ?: nexusRepo.nexusUrl.get()
+            descriptorRegistry.invalidateLater(nexusRepo.name, artifactRepo)
+            descriptorRegistry.tryGet(nexusRepo.name)?.stagingRepositoryUrl ?: URI(nexusRepo.nexusUrl.get().toString() + "/breakme/")
```

Comment:
> "/breakme/"

Has it left from your tests?

It would mean that we don't have a test to cover not set `stagingRepositoryUrl` :-/.

Reply:
This was part of my debugging to figure out wether this is related, but good point about manually setting the staging repo url to null or not setting it at all, in that case we would try to use the nexusUrl as fallback